src/ui/App.swift

```diff
@@ -257,7 +257,7 @@ class App: AppCenterApplication, NSApplicationDelegate {
     private func refreshSpecificWindows(_ windowsToUpdate: [Window]?, _ currentScreen: NSScreen) -> ()? {
         windowsToUpdate?.forEach { (window: Window) in
             guard appIsBeingUsed else { return }
-            window.refreshThumbnail()
+            if !Preferences.hideThumbnails { window.refreshThumbnail() }
```

Comment:
Hello. I'm not sure this is enough to avoid the sonoma screen sharing icon. I think Window.refreshThumbnail and Window.getPreview both should check Preferences.hideThumbnails before executing their code. This should completely guarantee not screenshotting the contents on other windows if the preference is set to true.

Reply:
I haven't seen the icon since..but maybe?

Reply:
@lawand thumbnails and previews are 2 different features. Users may want to disable one or the other. People who don't want any screen recording warning can disable both. I think status quo is correct